### DISCUSSION TITLE
Query Builder: Table and Column names in queries are not escaped properly

### DISCUSSION THREAD

**Discussion by tkorkunckaya** [grafana/grafana#37017] [Data Sources]
**What happened**:
Getting an SQL error: Error 1064: You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near 'order
WHERE
  date_created BETWEEN FROM_UNIXTIME(1576722724) AND FROM_UNIXTIME(1' at line 4
**What you expected to happen**:
Properly escaping table and column names when selecting from the builder.
**How to reproduce it (as minimally and precisely as possible)**:
Select a table named order (or any reserved word named table like rank, group etc.) from MySQL source.
**Anything else we need to know?**:
Please do not suggest writing the query manually, i already know how to do that.
**Environment**:
- Grafana version: 6.4.4
- Data source type & version: MySQL 8+ (applicable to any version)
- OS Grafana is installed on: Debian 8
- User OS & Browser: Windows 10, Chrome
- Grafana plugins: N/A
- Others:

**Comment by tkorkunckaya** [comment]
Hi,
I don't know why the issue is closed but i just wanted to update current situation, it still exists in Grafana v8.0.6 (68fe9e3431).
Here is the query created by the query builder: 
```
SELECT
  date_created AS "time",
  order_id
FROM order
ORDER BY date_created
```

Here is the error i got:

`db query error: Error 1064: You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near 'order ORDER BY date_created' at line 4`

Reproducing the error is simple, create a mysql data source and name the mysql table as any reserved word like "order". Query builder does not escape table names with proper SQL syntax using " ` ". In order to fix this issue every table and field name should be escaped in any database datasource while using query builder.

![image](https://user-images.githubusercontent.com/821861/126796752-4c8a2854-376e-46bf-a55e-182006105320.png)

The quick workaround i use is explicitly writing table names with back quotes. In this case the user should know SQL, and the bug and do not use UI's table name drop down selection or field name selection.

![image](https://user-images.githubusercontent.com/821861/126797306-b716e437-07b7-46ae-947f-c312d4a31cca.png)

Thanks.

**Comment by gabor** [comment]
hi @tkorkunckaya , this is fixed in the current grafana version, i was able to query a table named `time` in the visual query builder.
if you find the currrent solution insufficient, please create a new github issue and describe why the current solution is insufficient. thanks!

**Reply by tkorkunckaya** [reply]
Thanks Gabor,
I've confirmed with 10.2.1.
I couldn't find the actual commit but it seems to be limited to reserved words like 'order, sort, time' etc. In the drop down if the table name is a reserved word it lists with ticks (`order`). I'd like to check the code commit, but as I said I could not find. Checked the "datasources" project also. I did not checked with column names.
Thanks again.

**Reply by gabor** [reply]
i'm glad it worked. i also cannot point to the exact commit that fixed it, i know it got fixed between grafana8 and grafana9, so probably when the query-editor got rewritten.
